### DISCUSSION TITLE
Adding checkBoxes to Setup

### DISCUSSION THREAD

**Discussion by SonaBIMDev** [wixtoolset/issues#7237] [Questions]
Hi.
I'm using WixToolSet with VisualStudio.
I've got a Solution with myProject and the WixToolSetSetup project.
I've got multiple configuration of build in ConfigurationManager.
For example, I've got 4 configurations : 2020,2021,2022 and 2023.
For each configuration, a dll is builded for myProject. 
When I choose for exemple the 2023 configuration, MyProject is builded and a WixToolSetSetup est builded unsing this dll version.
Is it possible to generate a Setup wich can contains multiple CheckBoxes for each versions and the user can for example choose to install the files depending on the Version he choose?
For example, when the user choose to install the plugin for 2022, dll and other files are installed in a specific directory of the application using the plungin. 
See what I'm trying to do on the image bellow..
Many thx!

![screenshot_377](https://user-images.githubusercontent.com/54843266/220379151-15162aa3-0569-421e-9a29-6fff2c0e0a94.png)

**Comment by robmen** [comment]
Yes.

**Reply by SonaBIMDev** [reply]
Hi @robmen Thx for your reply
How can I do that?
I did not find some exemple ...

**Reply by robmen** [reply]
There are several ways to go about designing such a solution. It depends on exactly what you want to deliver, how you want to deliver it, and how you want to update it. The desired update story can usually be very informative to the design.

**Reply by SonaBIMDev** [reply]
@robmen 
As I wrote before, in my VS Solution, I've got multiple configurations : 2020,2021,2022,2023.
For each configuration, a dll is builded for myProject and a specific version of external references are used.
When I choose for exemple the 2023 configuration, MyProject is builded and a WixToolSetSetup est builded unsing the dll of MyProject.
But I'd like to get only one WixSetup and proposing the user to install differents dll version depending of versions user has checked.
The design of UI is not very important, WixToolSet design is cool and I'd like to continue using WixToolSet.
I saw how to do a CustomUI but, I'd like to get exemple of how propose multiple checkboxes etc. 
I saw this excellent presentation [(https://www.youtube.com/watch?v=usOh3NQO9Ms)] but no way to find solution to my problem...

**Reply by robmen** [reply]
Yeah, I read all that, but it wasn't clear where you wanted to start: a bundle or an MSI package?

What you want to do is definitely possible in a few different ways. It just isn't clear to me from the information you've provided what the best option is for your scenario. If you wanted to sit down and design a solution we could do that, but I don't have something for you to just cut and paste.

**Reply by SonaBIMDev** [reply]
I thought it would be possible to find with a little help a way to use WixToolSet to generate my own Setup. What you are proposing is to use your services for a fee, isn't it?

**Reply by robmen** [reply]
My company absolutely provides [commercial services](https://www.firegiant.com/services/). When working with customers, we make sure we understand the complete picture before making recommendations to ensure the customer starts off in the right direction. Of course, you are not obligated to use those services at all.

Personally, here in the OSS project, my focus is on improving the WiX Toolset's code and clarifying what is, what is not, or what might in the future be possible with the WiX Toolset. I am not the person on this Discussion forum that provides individual step-by-step guidance (I do enough of that with my customers during the day 😄).

You have [read the tutorial](https://www.firegiant.com/wix/tutorial/), right? It has two sections on customizing MSI UI. Unfortunately, it does not (yet) have information on customizing Bundle UI.

**Comment by SonaBIMDev** [comment]
Hi @robmen 
Thx for your answer.
I clearly understand what you do on your company, unfortunately, my company do not have enouth money to get your services even if I'll really like. 
Anyway, I'm gonna try to find a solution by myself. I've already gone through the tutorial but I didn't find the information I was looking for.

**Reply by robmen** [reply]
Understood. It is often a question of time vs money for all of us. Passion plays a part sometimes too.

**Reply by barnson** [reply]
Please don't post links to pirate sites. The book is available from its publisher: https://www.packtpub.com/product/wix-36-a-developers-guide-to-windows-installer-xml/9781782160427

**Reply by robmen** [reply]
Oh... and all the links we've provided are listed on the [Get help page](https://wixtoolset.org/docs/gethelp/).

**Comment by SonaBIMDev** [comment]
thx for all @robmen 
Sorry for the pirtate links, I did not realize it.
